libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -107,6 +107,52 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::copyToMemoryLiteralFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_memory_" + h256(_literal, h256::AlignLeft).hex();
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>() -> memPtr {
+				memPtr := <allocationFunction>(add(<size>, 32))
+				mstore(memPtr, <size>)
+				<storeLiteralInMem>(add(memPtr, 32))
+			}
+			)")
+			("functionName", functionName)
+			("allocationFunction", allocationFunction())
+			("size", to_string(_literal.size()))
+			("storeLiteralInMem", storeLiteralToMemoryFunction(_literal))
+			.render();
+	});
+}
+
+string YulUtilFunctions::storeLiteralToMemoryFunction(string const& _literal)
+{
+	string functionName = "store_literal_to_memory_" + h256(_literal, h256::AlignLeft).hex();
```

Comment:
Same here.

Reply:
Can you check with the name simplifier that the last part is properly cut off here if it is unambiguous?